app/src/main/java/bg/zahov/app/ui/history/HistoryFragment.kt

```diff
@@ -8,43 +8,42 @@ import android.view.MenuItem
 import android.view.View
 import android.view.ViewGroup
 import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.ui.platform.ComposeView
 import androidx.core.os.bundleOf
 import androidx.core.view.MenuProvider
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
-import androidx.lifecycle.map
 import androidx.navigation.fragment.findNavController
-import androidx.recyclerview.widget.LinearLayoutManager
-import bg.zahov.app.data.model.Workout
-import bg.zahov.app.data.model.state.HistoryUiMapper
-import bg.zahov.app.hideBottomNav
 import bg.zahov.app.setToolBarTitle
 import bg.zahov.app.showBottomNav
 import bg.zahov.app.showTopBar
 import bg.zahov.fitness.app.R
-import bg.zahov.fitness.app.databinding.FragmentHistoryBinding
 
 class HistoryFragment : Fragment() {
-    private var _binding: FragmentHistoryBinding? = null
-    private val binding
-        get() = requireNotNull(_binding!!)
-
     private val historyViewModel: HistoryViewModel by viewModels()
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?,
     ): View {
-        _binding = FragmentHistoryBinding.inflate(inflater, container, false)
-        requireActivity().showTopBar()
-        requireActivity().showBottomNav()
-        return binding.root
+        activity?.showTopBar()
+        activity?.showBottomNav()
+        setupTopBar()
+        return ComposeView(requireContext()).apply {
+            setContent {
+                HistoryScreen(historyViewModel) {
+                    findNavController().navigate(
+                        R.id.history_to_history_info,
+                        bundleOf(WORKOUTID to it)
+                    )
+                }
+            }
+        }
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        requireActivity().setToolBarTitle(R.string.history)
+    private fun setupTopBar() {
+        activity?.setToolBarTitle(R.string.history)
```

Comment:
For now this is okay, but the app should migrate to the Compose App bar at some point.